java/server/src/org/openqa/selenium/grid/sessionqueue/local/LocalNewSessionQueue.java

```diff
@@ -100,6 +113,36 @@ public int getQueueSize() {
     }
   }
 
+  @Override
+  public Map<String, Object> getQueueContents() {
+    Lock readLock = lock.readLock();
+    readLock.lock();
+    try {
+      List<Capabilities> capabilitiesList = sessionRequests.stream()
+        .map(SessionRequest::getHttpRequest)
+        .map(req -> {
+          try (
+            Reader reader = reader(req);
+            NewSessionPayload payload = NewSessionPayload.create(reader)) {
+            return payload.stream().iterator();
+          } catch (IOException e) {
+            LOG.warning("IOException while mapping to capabilities");
+          }
+          return null;
+        })
+        .filter(Objects::nonNull)
+        .filter(Iterator::hasNext)
+        .map(Iterator::next)
+        .collect(Collectors.toList());
+
+      return ImmutableMap.of(
+        "request-count", capabilitiesList.size(),
```

Comment:
If we are also adding the count, should we only leave this new endpoint and remove the one with the size?

Reply:
+1